### DISCUSSION TITLE
How to merge pull requests?

### DISCUSSION THREAD

**Discussion by LyndonYang1024** [oceanbase/oceanbase#60] [Q&A] [answered]
1.   Merge PR的时候可以只有正常的那个commit msg记录，不要出现merge branch/merge pr这种commit吗？这样不利于查看提交记录
2.  commit msg里面最好统一一下格式，最好可以带上issue号，这样后续再看这个提交的时候可以很清楚的知道这个提交的由来

**Comment by JingsongLi** [comment]
+1 to use:
- squash and merge
or
- rebase and merge

No use of merge commit

**Comment by yangbodong22011** [comment, marked as answer]
+1 for `squash and merge`
p.s. refer redis commit log: https://github.com/redis/redis/commits/unstable

**Reply by MizuhaHimuraki** [reply]
After further discussion, we decide to merge following pull requests in the 'squash and merge' way.

**Comment by MizuhaHimuraki** [comment]
We always merged pull requests by creating "merge commits" during the development of OceanBase in the past.

Because we have lots of engineers working for OceanBase kernel at the same time. We need to process hundreds of merge requests every week.

In this case, it is not appropriate to ask every pull request to be rebased to the latest code base before merging.

The squash merging will lose the details in the commit. So, this is a very bad choice in a project with millions of LOCs, too.

Therefore, we required developers to remove meaningless commits by using `git rebase -i HEAD~x`, and merge their pulls by creating a merge commit.
